### PR TITLE
test(show): recategorize and clean up browser tests

### LINKED ISSUE #7249: Recategorize and clean up browser tests for the `show` component
## Description

Recategorize, prune, and consolidate the browser tests in `packages/react/src/components/show` according to the rule from #7139.

## Files in scope

- `packages/react/src/components/show/show.test.browser.tsx`

## Work to do

Follow these steps in order. Skipping the baseline step makes any deletion claim unjustified.

1. **Capture baseline coverage** for this folder before changing anything. Run both projects (v8 coverage cannot mix jsdom and browser in one run) and record line / branch / function / statement coverage per source file:

   ```sh
   pnpm react test:coverage:jsdom --coverage.include='src/components/show/**' --coverage.reporter=text src/components/show/
   pnpm react test:coverage:chromium --coverage.include='src/components/show/**' --coverage.reporter=text src/components/show/
   ```

   Combine the two reports per source file (a line covered in either project counts as covered). This is the baseline.

2. **Re-categorize each test file** using the five-axis checklist in [test-categorization.md](https://github.com/yamada-ui/yamada-ui/blob/main/.agents/rules/test-categorization.md) (DOM read / event handling / observer / focus / browser API). Tests that do not need a real browser move to `*.test.{ts,tsx}` and `#test`. Tests that branch on engine should use `*.test.{chromium,firefox,webkit}.{ts,tsx}`.
   - **`a11y(...)` checks belong in jsdom** unless the component itself reads the real DOM. `a11y` is exported from both `#test` and `#test/browser`; default to the `#test` (jsdom) version because the assertion is a pure render + axe pass.
   - Re-categorization alone must not change combined coverage.

3. **Verify each removal candidate empirically.** A test is "redundant" only if removing it does not lower line, branch, function, or statement coverage on any source file in either project. To verify: temporarily delete the test, re-run the coverage command for the project that owns it, and compare to baseline. Do not delete by inspection alone — `displayName`, `className`, `role`, and `aria-*` assertions can each be the sole hit on a specific source line.

4. **Remove only the empirically verified tests** and consolidate scattered tests inside the same file.

5. **Re-run final coverage** with both commands from step 1. Confirm combined coverage on every source file is at least equal to baseline. Paste the baseline and final numbers in the PR description.

## Acceptance criteria

- `pnpm react test:jsdom --run src/components/show/` passes
- `pnpm react test:browser --run src/components/show/` passes
- `pnpm react lint` passes (no `#test` ↔ `#test/browser` boundary violations)
- Per-source-file combined coverage (line / branch / function / statement) on `packages/react/src/components/show/` is at least equal to the baseline captured in step 1, with both numbers posted in the PR description

Depends on #7139. Sub-issue of #7141.

## AI used

- [ ] I did not use AI to create this issue.
- [x] (If there is no check above) I checked the generated content before submitting.

### PR DESCRIPTION
**Pull request by hirotomoyamada**
Closes #7249

## AI used

- [ ] I did not use AI to create this PR.
- [x] (If there is no check above) I checked the generated content before submitting.

## Description

Recategorize, prune, and consolidate the browser tests in `packages/react/src/components/show` according to the rule introduced in #7139.

## Current behavior (updates)

One `*.test.browser.tsx` file lived under `packages/react/src/components/show/`:

- `show.test.browser.tsx` — 6 tests, all jsdom-friendly. Each test only renders `<Show>` and asserts `getByText` / `not.toBeInTheDocument` against text nodes. None of them read layout, dispatch real events, observe, manage focus, or use a browser API, so none require a real browser.

Two of those tests exercised the same `!when && no fallback` code path (`when={false}` and `when={null}`) and one exercised `!when && fallback` redundantly with a sibling, so removing them did not change any coverage axis.

## New behavior

Each test was re-categorized using the five-axis checklist in [`test-categorization.md`](.agents/rules/test-categorization.md).

**jsdom (`#test`)**

- `show.test.tsx` — 4 tests
  - `renders component correctly` (string child, `when` truthy → renders children)
  - `renders fallback when when is false` (falsy `when` with explicit fallback)
  - `renders children with function pattern` (`runIfFn` branch with object `when`)
  - `renders children with Element` (only test exercising the `isValidElement(result)` true branch)

**browser (`#test/browser`)**

- `show.test.browser.tsx` is removed because no browser-only assertion remained.

Removed tests that did not contribute to coverage (verified by deleting them and re-running `test:coverage:jsdom`):

- `renders nothing when when is false and no fallback is provided` — the `!when` branch with undefined fallback is the same code path covered by `renders fallback when when is false`.
- `renders nothing when when is null and no fallback is provided` — duplicate of the case above; both `false` and `null` go through the same `if (!when)` branch and produce the same `result = fallback` assignment.

## Is this a breaking change (Yes/No):

No. Test-only change. Public API of `Show` is unchanged.

## Additional Information

Verified locally:

- `pnpm react test:jsdom --run src/components/show/` — 4 tests pass
- `pnpm react test:browser --run src/components/show/` — no test files (the browser file was deleted because every assertion is jsdom-friendly)
- `pnpm react lint` — 0 warnings, 0 errors

Coverage on `packages/react/src/components/show/show.tsx` (combined across jsdom + browser projects):

| Project           | Source     | Stmts (before → after) | Branch (before → after) | Funcs (before → after) | Lines (before → after) |
| ----------------- | ---------- | ---------------------- | ----------------------- | ---------------------- | ---------------------- |
| jsdom (v8)        | `show.tsx` | 0% → 100%              | 0% → 100%               | 0% → 100%              | 0% → 100%              |
| chromium (istanbul) | `show.tsx` | 100% → n/a (no tests)  | 100% → n/a              | 100% → n/a             | 100% → n/a             |
| **Combined**      | `show.tsx` | **100% → 100%**        | **100% → 100%**         | **100% → 100%**        | **100% → 100%**        |

Combined per-source-file coverage matches the baseline.

Sub-issue of #7141.

### CODE REVIEW

**Review by claude-for-yamada-ui — DISMISSED** [review]

File: packages/react/src/components/show/show.test.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+  test("renders fallback when when is false", () => {
+    render(
+      <Show fallback="World" when={false}>
+        Hello
+      </Show>,
+    )
+
+    expect(screen.getByText("World")).toBeInTheDocument()
+  })
```
Comment (codex-for-yamada-ui) [inline packages/react/src/components/show/show.test.tsx:24]: **Issue:** This migration removes all coverage for the `fallback`-omitted falsy path (`when={false}` / `when={null}` with no `fallback`). In `Show`, `if (!when) result = fallback` can return `undefined`, and that “renders nothing” contract is no longer asserted anywhere in the current diff.

**Fix:** Add at least one jsdom test in this file for `<Show when={false}>Hello</Show>` (or `when={null}`) without `fallback`, and assert absence via `queryByText` + `not.toBeInTheDocument()`.

**Review by codex-for-yamada-ui — CHANGES_REQUESTED** [review]
- Test: This migration removes all coverage for the `fallback`-omitted falsy path (`when={false}` / `when={null}` with no `fallback`). In `Show`, `if (!when) result = fallback` can return `undefined`, and that “renders nothing” contract is no longer asserted anywhere in the current diff.

File: packages/react/src/components/show/show.test.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+  test("renders children with Element", () => {
+    render(
+      <Show fallback="World" when>
+        <Text>Hello</Text>
+      </Show>,
+    )
+
+    expect(screen.getByText("Hello")).toBeInTheDocument()
```
Comment (composer-for-yamada-ui) [inline packages/react/src/components/show/show.test.tsx:51]: **Issue:** Migrating from `show.test.browser.tsx` to jsdom drops two contracts that were explicitly asserted before: with no `fallback`, `when={false}` and `when={null}` must not render the `"Hello"` child (`Show` sets `result` to `undefined` when `!when` and `fallback` is absent; see `show.tsx` `if (!when) { result = fallback }`). The new suite ends at the Element children case and never asserts those empty-render paths, so the recategorization introduces a coverage regression for branches the old file covered.

**Fix:** Add two jsdom tests (for example using `render` + `expect(screen.queryByText("Hello")).not.toBeInTheDocument()`) for `<Show when={false}>Hello</Show>` and `<Show when={null}>Hello</Show>` without `fallback`, matching the removed browser tests.

**Review by composer-for-yamada-ui — CHANGES_REQUESTED** [review]
- Tests: Migrating from `show.test.browser.tsx` to jsdom drops two contracts that were explicitly asserted before: with no `fallback`, `when={false}` and `when={null}` must not render the `"Hello"` child (`Show` sets `result` to `undefined` when `!when` and `fallback` is absent; see `show.tsx` `if (!when) { result = fallback }`). The new suite ends at the Element children case and never asserts those empty-render paths, so the recategorization introduces a coverage regression for branches the old file covered.

File: packages/react/src/components/show/show.test.tsx (hunk 1 above)
Comment (codex-for-yamada-ui) [reply packages/react/src/components/show/show.test.tsx:24]: Fixed.

File: packages/react/src/components/show/show.test.tsx (hunk 2 above)
Comment (codex-for-yamada-ui) [reply packages/react/src/components/show/show.test.tsx:51]: Fixed.

**Review by claude-for-yamada-ui — APPROVED** [review]

**Review by codex-for-yamada-ui — APPROVED** [review]

**Review by composer-for-yamada-ui — APPROVED** [review]
